### PR TITLE
fix: `jsonpchunkloading` add preload script

### PR DESCRIPTION
**Pull request by caoren**
**What kind of change does this PR introduce?**

If a chunk is loaded，`__webpack_require__.f.j` method will time out.
Because `webbackjsonpcallback` does not process items in 'chunkloadingglobal'.
The pr fixed.

### CODE REVIEW

**Review by evilebottnawi — CHANGES_REQUESTED** [review]
Please accept CLA and add tests

File: lib/web/JsonpChunkLoadingRuntimeModule.js [hunk 1]
```diff
@@ -486,7 +486,9 @@ class JsonpChunkLoadingRuntimeModule extends RuntimeModule {
 						"",
 						`var chunkLoadingGlobal = ${chunkLoadingGlobalExpr} = ${chunkLoadingGlobalExpr} || [];`,
 						"var parentChunkLoadingFunction = chunkLoadingGlobal.push.bind(chunkLoadingGlobal);",
-						"chunkLoadingGlobal.push = webpackJsonpCallback;"
+						"chunkLoadingGlobal.push = webpackJsonpCallback;",
+						"chunkLoadingGlobal = chunkLoadingGlobal.slice();",
+						"for(var i = 0; i < chunkLoadingGlobal.length; i++) webpackJsonpCallback(chunkLoadingGlobal[i]);"
```
Comment (theKashey) [inline lib/web/JsonpChunkLoadingRuntimeModule.js:491]: a very missing feature. Blocking proper support for code splitting in `loadable-components` as long as right now or main chunk has to be loaded first, or the application will be broken.

**Comment by ScriptedAlchemy** [comment]
@sokra bumping this issue for visibility.

File: declarations.d.ts [hunk 2]
```diff
@@ -351,6 +351,25 @@ declare module "webpack-sources" {
...
+				  }
+				| {
+						path: string;
+						env?: string;
+				  }
+		): string | undefined;
+		export function findConfig(path: string): Record<string, string[]>;
+	}
+	export = browserslist;
+}
```
Comment (evilebottnawi) [inline declarations.d.ts:371]: Why do not use `@types/browserslist`?

**Review by evilebottnawi — COMMENTED** [review]
Also need rebase

**Comment by ScriptedAlchemy** [comment]
@caoren can you rebase :)

**Comment by 7rulnik** [comment]
Seems it was done in https://github.com/webpack/webpack/commit/436be4a029dd37e33f787039be53b9b5e8168026

**Pull request closed without merging**